_rules/table-header-cells-refer-to-data-cells.md

```diff
@@ -0,0 +1,160 @@
+---
+name: All table header cells refer to data cells
+
+test_type: atomic
+
+description: |
+  This rule checks that each table header in a data table refers to data cells.
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Jey Nandakumar
+- Audrey Maniez
+---
+
+## Test Procedure
+
+### Applicability
+
+This rule applies to any element with [semantic role](#semantic-role) of `columnheader` or `rowheader` within the `table` element that is either [visible](#visible) or [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+### Expectation
+- Each target element with the `headers` attribute refers to other `cells` of the same `table`.
+
+Each target element refers to data `cells` of the same `table`.
+
+## Assumptions
+
+*There are currently no assumptions*
+
+## Accessibility support
+
+*There are no major accessibility support issues known for this rule.*
```

Comment:
There are significant accessibility support issues with tables. Last time I tested (which is a few years ago, admittedly) none of the AT/UA fully implemented all table header mechanisms (header, scope, auto-scope)